server/app/services/UserService.ts

```diff
@@ -64,7 +71,55 @@ export class UserService {
     if (!updatedUser) {
       throw new NotFoundError(`User with username ${username} was not found`);
     }
-
     return UserEntity.buildUser(updatedUser);
   }
+
+  public async forgotPassword(url: string, body: IUser): Promise<string> {
+    const email = body.email;
+    const user: IUserEntity = UserEntity.buildUser(await this.userRepo.findByEmail(body));
+    const accessToken = generateToken(user);
+    requestMap.set(email, accessToken);
+
+    const info: SentMessageInfo = await this.systemService.sendEmail({
+      to: [email],
+      subject: 'ERP Password Reset',
+      emailBody: `This is the ERP Team07,\n\nPlease reset your password using the following link: ${url}/reset/${accessToken}\n\nRegards.`,
+    });
+    return info;
+  }
+
+  public async resetPassword(token: string, pass: string) {
+    let email = '';
+    let found = false;
+
+    try {
+      jwt.verify(token, config.get<string>('jwt.secret'), function (
+        err: jwt.JsonWebTokenError | jwt.NotBeforeError | jwt.TokenExpiredError | null,
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        decoded: any | IUserEntity
+      ) {
+        if (err) throw new BadRequestError('Invalid jwt token');
+
+        if (!(decoded.username && decoded.email && decoded.id && decoded.role))
+          throw new BadRequestError('Invalid jwt token');
+        delete decoded.exp;
+        delete decoded.iat;
+
+        email = decoded.email;
+        found = true;
+      });
+    } catch (err) {
+      return err;
+    }
```

Comment:
I feel like there is no point of this try catch. You can just throw and let the controller handle the error.

Reply:
its removed